content/blog/build-embedded-swift-application-for-esp32c6/_index.md

```diff
@@ -0,0 +1,125 @@
+---
+
+## Introduction
+
+Embedded Swift brings the power and expressivity of the Swift programming language to constrained environments such as microcontrollers and other embedded systems. Announced during [WWDC24](https://developer.apple.com/videos/play/wwdc2024/10197/), Embedded Swift aims to provide a lightweight, efficient subset of Swift that maintains the language's core features while significantly reducing its footprint and dependencies. This makes Swift a viable option for developers working in resource-constrained environments, allowing them to leverage Swift's modern syntax and safety features in new and exciting ways.
+
+## What is Embedded Swift?
+
+Embedded Swift is a specialized compilation mode of Swift designed to produce small, freestanding binaries suitable for embedded systems and bare-metal programming. It strips down unnecessary features and focuses on essential functionality to generate minimal and efficient code. Key goals include eliminating the large code size associated with the full Swift runtime, simplifying the generated code, and allowing effective dead-code stripping.
+
+Embedded Swift is not a complete SDK or HAL; rather, it provides the tools to compile Swift code into object files that can be integrated with existing embedded development workflows. It supports environments with and without a dynamic heap and aims to remain a subset of Swift, ensuring code compatibility with regular Swift projects.
+
+## Getting Started with Swift for ESP32-C6
+
+The following example covers steps for building Embedded Swift application for ESP32-C6.
+
+### Hardware Setup
+
+Ensure you have the following hardware:
+
+- [ESP32-C6](https://www.espressif.com/en/products/socs/esp32-c6) development board.
+- USB-C Cable for power and programming.
+
+### Prerequisites
+
+Before you begin, make sure you have the following:
+
+- [ESP-IDF v5.3](https://docs.espressif.com/projects/esp-idf/en/release-v5.3/esp32/get-started/index.html): The official development framework for the ESP32, properly installed and sourced in your shell.
+- [Swift 6](https://www.swift.org/install/macos/#development-snapshots)
+
+### Building an Example Project
+
+First, let's see the whole build process in Asciinema recording:
+{{< asciinema key="build-swift-app-for-esp32c6" cols="80" rows="24" poster="npt:0:08" >}}
+
+1. **Clone Sample Project**:
+
+   Let's clone the project and configure `TOOLCHAINS` environment variable with the version of the installed Swift 6.
+
+   ```bash
+   git clone git@github.com:apple/swift-embedded-examples.git --single-branch --branch main
+   export TOOLCHAINS=org.swift.600202406111a
+   cd swift-embedded-examples/esp32-led-strip-sdk
+   ```
+
+2. **Set the Target**:
+
+   Ensure the correct target is set for your project:
+
+   ```bash
+   idf.py set-target esp32c6
+   ```
+
+3. **Build and Flash the Project**:
+
+   Compile and flash your application to the ESP32-S3-BOX-3:
```

Comment:
Up above, the required hardware is the ESP32-C6 development board. Here, it transformed to the ESP32-S3-BOX-3. Shouldn't it be the same hardware?